### PR TITLE
Task 6

### PR DESCRIPTION
**Pull request by nmparfyonov**
FE link: https://d2eoo74ecvbfun.cloudfront.net/
Task: https://github.com/rolling-scopes-school/aws/blob/main/aws-developer/06_async_microservices_communication/task.md
[test.csv](https://github.com/nmparfyonov/nodejs-aws-shop-backend/files/13628868/test.csv)

### CODE REVIEW

File: product-service/test/catalogBatchProcess.test.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
+const { handler } = require('./catalogBatchProcess');
```
Comment (Jimmba) [inline product-service/test/catalogBatchProcess.test.ts]: You don't have a file 'catalogBatchProcess' in test folder. What do you import? Are you sure that your tests work? Did you try to run them?

File: product-service/test/catalogBatchProcess.test.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+                        title: 'Sample Product',
+                        description: 'Product Description',
+                        price: 30,
+                        count: 100,
+                    }),
+                },
+            ],
+        };
+
+        const snsPublishMock = jest.spyOn(sns, 'publish').mockReturnValue({ promise: jest.fn() });
```
Comment (Jimmba) [inline product-service/test/catalogBatchProcess.test.ts]: You try to spy function 'sns', but you are not even importing it. I guess you didn't run this test. It doesn't work.

File: product-service/resources/catalogBatchProcess/catalogBatchProcess.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+const sns = new AWS.SNS();
+const dynamoDB = new AWS.DynamoDB.DocumentClient();
+
+exports.handler = async (event) => {
+    try {
+        const { Records } = event;
+
+        if (!Records || Records.length === 0) {
+            console.log('No records found in the SQS event.');
+            return { statusCode: 200 };
```
Comment (Jimmba) [inline product-service/resources/catalogBatchProcess/catalogBatchProcess.js]: You should use 404 status code 'Not found' in this case

File: product-service/test/catalogBatchProcess.test.ts (hunk 1 above)
Comment (nmparfyonov) [reply product-service/test/catalogBatchProcess.test.ts]: Clone repo. Add catalogBatchProcess.js file to test folder and see gitignore rules.

**Pull request closed without merging**